src/index.js

```diff
@@ -1,13 +1,126 @@
+
+    checkUserInput(userInputNumbers) {
+        let userNumArray = userInputNumbers.split('');
+        if (userNumArray.length != 3) {
+            return false;
+        }
+        else if (isNaN(userInputNumbers) == true) {
+            return false;
+        }
+        let ret = true;
+        userNumArray.forEach(function(item, index) {
+            if (item == userNumArray[(index + 1) % 3] || item == userNumArray[(index + 2) % 3]) {
+                ret = false;
+            }
+        })
+        if (ret == false) {
+            return false;
+        }
+        return true;
+    }
+  
+    play(computerInputNumbers, userInputNumbers) {
+      let computerNumArray = this.computerNum.split('');
+      let userNumArray = userInputNumbers.split('');
+  
+      console.log(this.computerNum);
+      console.log(userInputNumbers);
+      computerNumArray.forEach(function(item, index) {
+        if (item == userNumArray[index]) {
+          this.strike += 1;
+        }
+        if (
+          item == userNumArray[(index + 2) % 3] ||
+          item == userNumArray[(index + 1) % 3]
+        ) {
+          this.ball += 1;
+        }
+      }, this);
+      if (this.ball == 0 && this.strike == 0) {
+        this.result = '낫싱';
+      } else if (this.ball != 0 && this.strike != 0) {
+        this.result =
+          this.ball.toString() + '볼 ' + this.strike.toString() + '스트라이크';
+      } else if (this.ball == 0 && this.strike != 0) {
+        this.result = this.strike.toString() + '스트라이크';
+      } else if (this.ball != 0 && this.strike == 0) {
+        this.result = this.ball.toString() + '볼';
+      }
+      if (this.result == '3스트라이크') {
+          this.result = '🎉 정답을 맞추셨습니다! 🎉';
+      }
+      return this.result;
+    }
+
+    clickCheck(event) {
+      let userInputNumbers = document.getElementById('user-input').value;
+      console.log(userInputNumbers);
+      let checkInput = baseball.checkUserInput(userInputNumbers);
+      console.log(checkInput);
+      if (checkInput == false) {
+          alert('유효하지 않은 입력입니다.')
+          document.getElementById('user-input').value = '';
+          return ; 
+      }
+      let resultText = baseball.play(baseball.computerNum, userInputNumbers);
+      let result = document.getElementById('result');
+      result.innerText = resultText;
+      baseball.ball = 0;
+      baseball.strike = 0;
+      baseball.checkSuccess(resultText);
+    }
+
+    checkSuccess(resultText) {
+      if (resultText == '🎉 정답을 맞추셨습니다! 🎉') {
+          console.log("haha")
+          let newDiv = document.createElement('div');
+          newDiv.innerHTML = '게임을 새로 시작하시겠습니까?';
+          newDiv.setAttribute('id', 'newDiv');
+          let newBtn = document.createElement('button');
+          newBtn.innerHTML = '게임 재시작';
+          newBtn.setAttribute('id', 'newBtn')
+          newDiv.append(newBtn);
+          document.body.appendChild(newDiv);
+          newBtn.addEventListener('click', this.restartGame);
+      }
+    }
```

Comment:
```suggestion
      if (resultText == '🎉 정답을 맞추셨습니다! 🎉') {
          console.log("haha")
          let newDiv = document.createElement('div');
          let newBtn = document.createElement('button');
          newDiv.setAttribute('id', 'newDiv');
          newBtn.setAttribute('id', 'newBtn')
          newBtn.innerHTML = '게임 재시작';
          newDiv.innerHTML = '게임을 새로 시작하시겠습니까?';
          newDiv.append(newBtn);
          document.body.appendChild(newDiv);
          newBtn.addEventListener('click', this.restartGame);
      }
    }
```
뭔가 이런것도 조금 더 짧아질 여지가 있는것 같아서 리뷰했습니다.